### DISCUSSION TITLE
Mapping of 2D Autocorrelation descriptors to atomic properties

### DISCUSSION THREAD

**Discussion by umesh-lantern** [rdkit/rdkit#6122] [Q&A]
Hi rdkit,

I am using rdkit to generate 2D autocorrelation descriptors and would like to understand the mapping. I found one slide (from this slide deck - https://github.com/rdkit/UGM_2017/blob/master/Presentations/Godin_3D_Descriptors.pdf) in rdkit github repo (attached screenshot) explaining about the Position and Atomic properties. Most of the properties are easy to understand, but was not able to figure out what the "Relative IState" means, which belongs to the position 41-48

![image](https://user-images.githubusercontent.com/52801831/220692396-89ffebfd-a109-4c46-8946-9063814e5f70.png)

Can you please describe what "Relative IState" means here, and also provide a reference?

Thanks

**Comment by thegodone** [comment]
Hi,

E-State are defined as atom features based on the documentations I used
when I implemented them.

I've normalized (rescaled) them to get the so called "Relative" E-State.
Based on book definitions (
https://onlinelibrary.wiley.com/doi/book/10.1002/9783527628766 p27: for an
example Moreau-Broto autocorrelation or ATS, p 283: for Electrotolpological
state indices p 285: there is a small sentence in the bottom: "The
electrotopological states are -> local vextex invariants. After rescaling,
they are also used as atomic weighting factors for the calculation of ->
WHIM descriptors"). Because those descriptors were available (cause I
implemented WHIM too), I reused them for others 2D/3D "descriptors".

You can find the related code :
https://github.com/rdkit/rdkit/blob/df2b6c8a8c775748c1dcac83af0f92af434bab81/Code/GraphMol/Descriptors/MolData3Ddescriptors.cpp#L127

Hope it helps.

Guillaume


Le mer. 22 févr. 2023 à 17:32, umesh-lantern ***@***.***> a
écrit :
